processing/src/main/java/org/apache/druid/query/monomorphicprocessing/SpecializationService.java

```diff
@@ -65,19 +69,162 @@
 {
   private static final Logger LOG = new Logger(SpecializationService.class);
 
-  private static final Unsafe UNSAFE;
+  private static final MethodHandle DEFINE_CLASS;
+  private static final RuntimeException DEFINE_CLASS_NOT_SUPPORTED_EXCEPTION;
 
   static {
+    MethodHandle defineClass = null;
+    RuntimeException exception = null;
     try {
-      Field theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
-      theUnsafe.setAccessible(true);
-      UNSAFE = (Unsafe) theUnsafe.get(null);
+      defineClass = lookupDefineClassMethodHandle();
     }
-    catch (Exception e) {
-      throw new RuntimeException("Cannot access Unsafe methods", e);
+    catch (RuntimeException e) {
```

Comment:
`lookupDefineClassMethodHandle` can be inlined in this method to be able to make `DEFINE_CLASS_NOT_SUPPORTED_EXCEPTION` of `UnsupportedOperationException` type

Reply:
I'm not sure there is much benefit in doing so.

Reply:
IMO `lookupDefineClassMethodHandle()` is not a good abstraction, having this method vs. not having it just complicates the understanding and reading the code. There is no separate method in `UnsafeUtil` in a similar situation.